packages/expo-updates/android/src/main/java/expo/modules/updates/UpdatesController.java

```diff
@@ -220,6 +224,16 @@ public boolean isEmergencyLaunch() {
     return mIsEmergencyLaunch;
   }
 
+  // internal setters
+
+  /* package */ void setSelectionPolicy(SelectionPolicy selectionPolicy) {
```

Comment:
It's counterintuitive to have the state of a setter get automatically cleared. We could do two things to address this: name this method something like `setNextSelectionPolicy` or `setSelectionPolicyOnce` to communicate that it will be used for only the next update, or we could make this method's caller be responsible for calling `resetSelectionPolicyToDefault`.

Reply:
Would it achieve the same purpose to expose `defaultSelectionPolicy()` rather than `resetSelectionPolicyToDefault()` and then call `setSelectionPolicy(defaultSelectionPolicy())` in the place we currently call `reset`? (I added `reset` simply to avoid having to thread multiple new methods through UpdatesInterface/Service/Binding/versioned copies, but the intent is to set to the default while letting this class control what the default is.)

Or does the counterintuitivity stem from just having two different external places that call any sort of setter?

Reply:
> Or does the counterintuitivity stem from just having two different external places that call any sort of setter?

Yeah. If the dev client calls setSelectionPolicy(X), it's natural to expect the selection policy to remain X until the dev client or some other user of expo-updates sets a new selection policy, rather than expo-updates itself clearing the policy.